### DISCUSSION TITLE
App.run interuptable

### DISCUSSION THREAD

**Discussion by jevylux** [miguelgrinberg/microdot#256] [Q&A]
One of the reasons i am looking for a replacement of the isage of phew, is once you start the server with phew.run(), you cannot do anything else.
Is it the same withh app.run()?
i often want to use an api at the same time as i use mqtt in the same application.
How could this be made possible?

**Comment by miguelgrinberg** [comment]
I don't use phew so I cannot answer questions on comparisons. The `app.run()` method in microdot is blocking, but it is intended to be used in cases where the is no other concurrent activity. If you want to run the web server in the background you can use `app.start_server`, which is an asynchronous function.

Example:

```python
            async def main():
                web_server_task = asyncio.create_task(app.start_server())
                exit = False
                while not exit:
                    # do whatever you want here (as long as it is non-blocking)
                    # set exit=True to exit
                await web_server_task
```
